Code/Framework/AzCore/AzCore/DOM/DomUtils.h

```diff
@@ -30,6 +31,29 @@ namespace AZ::Dom::Utils
     };
 
     bool DeepCompareIsEqual(const Value& lhs, const Value& rhs, const ComparisonParameters& parameters = {});
+    Value TypeIdToDomValue(const AZ::TypeId& typeId);
+    AZ::TypeId DomValueToTypeId(const AZ::Dom::Value& value, const AZ::TypeId* baseClassId = nullptr);
+    JsonSerializationResult::ResultCode LoadViaJsonSerialization(void* object, const AZ::TypeId& typeId, const Value& root, const JsonDeserializerSettings& settings = {});
+    JsonSerializationResult::ResultCode StoreViaJsonSerialization(const void* object, const void* defaultObject, const AZ::TypeId& typeId, Value& output, const JsonSerializerSettings& settings = {});
+
+    template <typename T>
+    JsonSerializationResult::ResultCode LoadViaJsonSerialization(T& object, const Value& root, const JsonDeserializerSettings& settings = {})
+    {
+        return LoadViaJsonSerialization(&object, azrtti_typeid<T>(), root, settings);
+    }
```

Comment:
Nitpick: Consider putting the implementation into a .inl to improve the readability of the header.

Reply:
I'm going to avoid tackling this now to preserve the diff, but that's a reasonable bit of future tidy-up (even if Visual Studio seems to lose all semblance of semantic understanding when a header includes its implementation for some reason).